### PR TITLE
Clean Up Makefile of Runtime

### LINKED ISSUE #1006: Refactor Makefile
The file of  `/blueoil/converter/templates/Makefile` should be refactored because it has some problem below.

* The header file changes should be detected
* Redundant code should be removed (same code is seen many times)
* Some bugs should be fixed

We need to fix them for future maintenance.

### PR DESCRIPTION
**Pull request by tk26eng**
## What this patch does to fix the issue.
* This patch cleans up Makefile of runtime
* This enables easier way to add the new supported architecture
* The options for make is also changed
* This add new feature and it can detect changed header file correctly

## Link to any relevant issues or pull requests.
Fix #1006

### CODE REVIEW

**Comment by tk26eng** [comment]
@primenumber 
@tfujiwar 
This code includes changes of Makefile and some python code.
Could you review this PR ?

File: blueoil/converter/templates/Makefile [hunk 1]
```diff
@@ -108,139 +87,108 @@ clean:
...
-	$(AR) $(LDFLAGS) libdlk_$(NAME).a $(LIB_OBJ) $(LIB_ARM_OBJ)
-
-$(ARS_FPGA): $(LIB_OBJ) $(LIB_FPGA_OBJ)
-	$(AR) $(LDFLAGS) libdlk_$(NAME).a $(LIB_OBJ) $(LIB_FPGA_OBJ)
+	-$(RM) $(ALL_OBJ)
+	-$(RM) $(ALL_DEP)
+
+
+# Base Settings
+FLAGS := -I./include -O3 -std=c++14 -g -MMD -MP
```
Comment (primenumber) [inline blueoil/converter/templates/Makefile:95]: `CXXFLAGS` is commonly used for C++ compile flags.

Comment (tk26eng) [reply blueoil/converter/templates/Makefile]: @primenumber 
The FLAGS variable has been replaced with CXXFLAGS instead

**Comment by tk26eng** [comment]
Fix #436

**Pull request merged**